dbus/dbus.hpp

```diff
@@ -26,29 +27,41 @@ class SDBusPlus
      * @param[in] bus - bus.
      * @param[in] busName - bus name.
      * @param[in] objPath - object path.
-     * @param[in] value - value to be set.
+     * @param[in] value - value to be set, in degrees, regardless of unitMilli.
+     * @param[in] unitMilli - true to set millidegrees, false to set degrees.
      */
-    static auto
+    static void
         setValueProperty(sdbusplus::bus::bus& bus, const std::string& busName,
-            const std::string& objPath, const int64_t& value)
+            const std::string& objPath, const double& value, bool unitMilli)
     {
-        std::variant<double> data = (double)value / 1000;
-
         try
         {
             auto methodCall = bus.new_method_call(
                 busName.c_str(), objPath.c_str(), dbusPropertyInterface, "Set");
 
             methodCall.append(valueInterface);
             methodCall.append("Value");
-            methodCall.append(data);
+
+            // Using millidegrees on D-Bus seems to need int64_t not double
+            if (unitMilli)
+            {
+                // Convert degrees to millidegrees
+                auto intValue = static_cast<int64_t>(value * 1000.0);
```

Comment:
I've been burned before by integer constants triggering unwanted conversion to doing integer arithmetic, so when there's a constant in an expression, I want to make sure it has a decimal point, so the compiler keeps it as double. That way, a value of 0.5 will be scaled up to 500 appropriately here, instead of being prematurely rounded off then scaled wrong.

Reply:
Oops, sry, I missed "const double& value" in the function signature. You are right about this.